src/extensions/default/InAppNotifications/main.js

```diff
@@ -0,0 +1,372 @@
+    }
+
+    function _checkExpiry(expiry) {
+        return Date.now() <= expiry;
+    }
+
+    function _checkExtensions(filters) {
+        var allExtensions = ExtensionManager.extensions,
+            allExtnsMatched = true,
+            userExtensionKeys = Object.keys(allExtensions).filter(function(k) {
+                return allExtensions[k].installInfo.locationType === 'user';
+            });
+
+        if (!filters.extensions) {
+            allExtnsMatched = userExtensionKeys.size === 0;
+        } else if (filters.extensions.length === 0) {
+            allExtnsMatched = userExtensionKeys.length > 0;
+        } else {
+            var filteredExtns = filters.extensions,
+                extnIterator = null;
+            for (var i=0; i < filteredExtns.length; i++) {
+                extnIterator = filteredExtns[i];
+                if (userExtensionKeys.indexOf(extnIterator) === -1) {
+                    allExtnsMatched = false;
+                    break;
+                }
+            }
+        }
+        return allExtnsMatched;
+    }
+
+    function _checkNotificationValidity(notificationObj) {
+
+        var filters = notificationObj.filters,
+            _platform = brackets.getPlatformInfo(),
+            _locale = brackets.getLocale(),
+            _lastHandledNotificationNumber = PreferencesManager.getViewState("lastHandledNotificationNumber"),
+            // Extract current build number from package.json version field 0.0.0-0
+            _buildNumber = Number(/-([0-9]+)/.exec(brackets.metadata.version)[1]),
+            _version = brackets.metadata.apiVersion;
+
+        if(_locale.length > 2) {
+            _locale = _locale.substring(0, 2);
+        }
+
+        return  notificationObj.sequence > _lastHandledNotificationNumber
+                && _checkExpiry(notificationObj.expiry)
+                && _checkPlatform(filters, _platform)
+                && _checkLocale(filters, _locale)
+                && _checkVersion(filters, _version)
+                && _checkBuild(filters, _buildNumber)
+                && _checkExtensions(filters);
+    }
+
+
+    /**
+     * Removes and cleans up the notification bar from DOM
+     */
+    function cleanNotificationBar() {
+        var $notificationBar = $('#notification-bar');
+        if ($notificationBar.length > 0) {
+            $notificationBar.remove();
+        }
+    }
+
+    /**
+     * Displays the Notification Bar UI
+     * @param   {object} msgObj - json object containing message info to be displayed
+     *
+     */
+    function showNotification(msgObj) {
+        var $htmlContent = $(msgObj.html),
+            $notificationBarElement = $(NotificationBarHtml);
+
+        // Remove any SCRIPT tag to avoid secuirity issues
+        $notificationBarElement.find('script').remove();
+
+        // Remove any STYLE tag to avoid styling impact on Brackets DOM
+        $notificationBarElement.find('style').remove();
```

Comment:
Shouldn't we be sanitising `$htmlContent` as this is the network response?

Reply:
Explanation mentioned before.

Reply:
Actually, I meant we should be sanitising `$htmlContent` variable instead of the `$notificationBarElement`.

Reply:
Done. Good catch!